OBAKit/Helpers/OBAStopIconFactory.m

```diff
@@ -16,38 +16,298 @@
+// placement, and the size and position of the direction chevron.
+
+#define IconCornerRadiusPercentage  0.1f    // Radius of icon corner, relative to overall size
+#define StrokeWidthPercentage       0.038f  // Width of stroke for all borders, relative to overall size
+#define IconInsetPercentage         0.21f   // Amount of padding to inset the icon (leaving room for chevron), relative to overall size
+#define GlyphAreaPercentage         0.7f    // Amount of icon height reserved for glyph
+#define TextAreaPercentage          (1.0f - GlyphAreaPercentage) // Amount of icon height reserved for text
+#define GlyphHeightPercentage       0.6f    // Height of glyph, relative to icon height
+#define GlyphInsetRelativeToStroke  1.5f    // Padding around glyph, relative to stroke width
+#define ChevronWidthPercentage      0.25f   // Width of chevron, relative to overall size (assuming N chevron)
+#define ChevronHeightPercentage     0.15f   // Height of chevron, relative to overall size (assuming N chevron)
+
+// For the simple getIconForStop call, here are the default dimensions
+#define StopIconDefaultWidth        54
+#define StopIconDefaultHeight       54
+
+// . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+
+static NSDictionary *rotationAngles;
+static UIColor *strokeColor;
+static UIColor *iconBackgroundColor;
+static UIColor *textBackgroundColor;
+static UIColor *chevronFillColor;
+static UIColor *textColor;
+static NSString *stopLabelFaceName;
+static NSString *stopLabelText;
+static NSCache *iconCache;
+
++(void)initialize {
+    iconCache = [[NSCache alloc] init];
+
+    stopLabelText = NSLocalizedString(@"STOP",);
+
+    // I chose "Helvetica" because it's the official typeface of the MTA
+    stopLabelFaceName = @"Helvetica Bold";
+
+    // Color selections
+    // TODO: Consider OBATheme here?
+    strokeColor = [UIColor blackColor];
+    iconBackgroundColor = [UIColor whiteColor];
+    textBackgroundColor = strokeColor;
+    chevronFillColor = [UIColor redColor];
+    textColor = [UIColor whiteColor];
+
+    // Rotation transform angles
+    rotationAngles = @{@"N"    : @(0.f),
+                       @"NE"   : @(0.7853f),
+                       @"E"    : @(1.5708f),
+                       @"SE"   : @(2.3562f),
+                       @"S"    : @(3.1416f),
+                       @"SW"   : @(3.9270f),
+                       @"W"    : @(4.7124f),
+                       @"NW"   : @(5.4978f)};
+}
+
 + (UIImage *)getIconForStop:(OBAStopV2 *)stop {
-    NSString *routeIconType = [self getRouteIconTypeForStop:stop];
-    NSString *direction = @"";
+    return [self getIconForStop:stop withSize:CGSizeMake(StopIconDefaultWidth, StopIconDefaultHeight)];
+}
 
-    if (stop.direction) {
-        direction = stop.direction;
++ (UIImage *)getIconForStop:(OBAStopV2 *)stop withSize:(CGSize)size {
+    // First, let's compose the cache key out of the name and orientation, then
```

Comment:
This is always called from a mapkit delegate method, so should forever be on thread 0. The only mutable state is the image cache, and NSCache is thread safe, so there should be no issue.